cmd/migrateblobs/main.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// This command migrates blobstore blobs of charms and resources for all
+// entities.  This command is intended to be run on the production db and then
+// discarded.  The first time this command is executed, all the entities are
+// updated.  Subsequent runs migrates missing destination blobs.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+
+	"github.com/juju/loggo"
+	"github.com/juju/utils/parallel"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/goose.v2/client"
+	"gopkg.in/goose.v2/errors"
+	"gopkg.in/goose.v2/identity"
+	"gopkg.in/goose.v2/swift"
+	"gopkg.in/mgo.v2"
+
+	"gopkg.in/juju/charmstore.v5-unstable/config"
+)
+
+var (
+	logger        = loggo.GetLogger("migrateblobs")
+	loggingConfig = flag.String("logging-config", "INFO", "specify log levels for modules e.g. <root>=TRACE")
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "usage: %s [options] <source config path> <destination config path>\n", filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Source config path is used to it read charmstore data. Destination config path is used only for writing new blobs.\n")
```

Comment:
Do we actually need both source and destination config paths?
We only use MongoURL from the source path, which needs to
be in the destination path too, so I think that just a single destination
config file is needed.